platform/wayland/cog-platform-wl.c

```diff
@@ -1819,6 +1651,30 @@ cog_wl_platform_create_im_context(CogPlatform *platform)
     return NULL;
 }
 
+void
+cog_wl_platform_popup_create(CogWlPlatform *platform, WebKitOptionMenu *option_menu)
+{
+    g_assert(!platform->popup);
+
+    CogWlPopup *popup = cog_wl_popup_create(platform, option_menu);
+    platform->popup = popup;
```

Comment:
What happens if a popup had already been created? Probably the correct thing to do is add a check, and if there was already a popup, close+destroy it before creating the new one:

```suggestion
    g_clear_pointer(&platform->popup, cog_wl_popup_destroy);
    platform->popup = cog_wl_popup_create(platform, option_menu);
```

Reply:
in the current implementation, this is not possible since there is an assertion in https://github.com/Igalia/cog/pull/635/files#diff-7342248ee4ad822cd294d9c8341a52fd10b3d1df4dd2a202ccb926df67d51034R1657. The current implementation relies in a logical management if the Popup by the Platform.

Reply:
but ensure the function is used in the expected way. Anyhow, this could be also expressed inside a documentation comment. I will do. Thanks

Reply:
Documentation added in https://github.com/Igalia/cog/pull/635/files#diff-dfb5db5df9086c64aff35fe3e3f49a702cfadab4aa381fa01e96ad29a52ed8ecR39